### PR TITLE
Templates for basic functionality

### PR DESCRIPTION
**Pull request by beneblubb**

### CODE REVIEW

File: src/environment.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
+from src.common_constants import CommonConstants
+
+class Environment():
+    def __init__(self) -> None:
+        self.pixel2meter = CommonConstants.PIXEL_TO_METER
```
Comment (Dummyrunner) [inline src/environment.py:5]: `self.pixel_to_meter_ratio` would help to distinguish it from the methods

File: src/game_window.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        self.fps = 60
+        self.is_running = True
+
+    def set_caption(self, title):
+        pygame.display.set_caption(title)
+
+    def erase_screen(self):
+        self.window.fill((0, 0, 0))
+
+    # Von Copilot vorgeschlagen:
```
Comment (Dummyrunner) [inline src/game_window.py:21]: remove this

File: src/game_window.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
+import pygame
+import src.common_constants as common_constants
+
+class GameWindow():
+    def __init__(self, title):
+        self.width = common_constants.CommonConstants.WINDOW_WIDTH
```
Comment (Dummyrunner) [inline src/game_window.py:6]: really long line! change import to `from common_constants import CommonConstants`, then you can refer to WINDOW_WITH as `CommonConstants.WINDOW_WIDTH`

File: src/global_physics.py [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import math
+import numpy as np
+import src.common_constants as common_constants
+
+class GlobalPhysics():
+    def __init__(self):
+        self.gravity = common_constants.CommonConstants.GRAVITATION
+        self.air_density = common_constants.CommonConstants.AIR_DENSITY
+        self.objects_assected_by_physics = []
```
Comment (Dummyrunner) [inline src/global_physics.py:9]:
```suggestion
        self.physical_objects = []
```
shorter, convenient for an object that will be used frequently

File: src/global_physics.py [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+
+    def handle_physics(self):
```
Comment (Dummyrunner) [inline src/global_physics.py:11]:
```suggestion
    def step_physics(self):
```

File: src/rocket.py [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+
+class Rocket(pygame.sprite.Sprite):
+    def __init__(self, x, y, width, height, color, velocity, acceleration):
+        pygame.sprite.Sprite.__init__(self)
+        self.image = pygame.Surface((width, height))
+        self.image.fill(color)
+        self.rect = self.image.get_rect()
+        self.rect.x = x
+        self.rect.y = y
+        self.velocity = velocity
```
Comment (Dummyrunner) [inline src/rocket.py:11]: velocity, acceleration should be inherited from `LinearKinematics`

File: src/rocket.py [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+        self.acceleration = acceleration
+
+    def move(self):
+        self.rect.x += self.velocity
+        self.velocity += self.acceleration
```
Comment (Dummyrunner) [inline src/rocket.py:16]:
```suggestion
```
move rect only horizontally? Independet from timestep?

Comment (Dummyrunner) [reply src/rocket.py:16]: I would prefer to have an extra position attribute and update rect accordingly. I feel the decoupling of physics and updating game mechanics is better then

**Review by Dummyrunner — CHANGES_REQUESTED** [review]

File: src/GlobalPhysics.py [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+class GlobalPhysics():
+    def __init__(self):
+        self.gravity = CommonConstants.GRAVITATION
+        self.air_density = CommonConstants.AIR_DENSITY
+        self.objects_assected_by_physics = []
+
+    def handle_physics(self):
+        pass
+
+    def collision_detection(self):
```
[the remark this answers is not in the document]
Comment (beneblubb) [reply src/GlobalPhysics.py:12]: Don't know. We wrote in the plan.md, that GlobalPhysics should have this methods. How did you assume that they will work?

File: src/rocket.py (hunk 6 above)
Comment (beneblubb) [reply src/rocket.py:11]: line this: `self.velocity = LinearKinematics(velocity)`?

Comment (Dummyrunner) [reply src/rocket.py:11]: class Rocket(bla.sprite, LinearKinematics):
...

Gives rocket all public and protected members that linearkinematics has. (Inheritance)

**Review by Dummyrunner — APPROVED** [review]

**Pull request merged**